packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -614,6 +614,12 @@ class DevFS {
       // before processing bundle.
       _logger.printTrace('Processing bundle.');
       // await null to give time for telling the compiler to compile.
+      // TODO(andrewkolos): This is a hack. Adding any more awaits to the compiler's
+      //  recompile method will cause this to be insufficent.
+      //  https://github.com/flutter/flutter/issues/151255.
+      await null;
```

Comment:
There is some shady looking stuff going on in this PR. I am skeptical about cp'ing this to stable.

Reply:
This was an extension of an existing hack (there was already one `await null`) here; but, yes, I agree there are definitely a few unshaven yaks when it comes to Dart (re)-compilation in the tool.

While not CP'ing this change could delay getting stack traces on https://github.com/flutter/flutter/issues/137184 (easily the top crasher in the tool), I certainly understand not wanting to CP this.

P.S. I saw https://github.com/flutter/flutter/issues/153026. Yes, hot reload has been flaky for a while. I'm doing some archeology to put a summary together here. However, if hot reload has become especially flaky (well over 2%) since this PR landed, I'm okay with reverting.